src/actions/requests.rs

```diff
@@ -618,37 +636,54 @@ impl<'a> Action<'a> for CodeAction {
 
 impl<'a> RequestAction<'a> for CodeAction {
     type Response = Vec<Command>;
-    fn handle<O: Output>(&mut self, _id: usize, params: Self::Params, ctx: &mut ActionContext, _out: O) -> Result<Self::Response, ()> {
+    fn handle<O: Output>(&mut self, _id: usize, params: Self::Params, ctx_: &mut ActionContext, _out: O) -> Result<Self::Response, ()> {
```

Comment:
Could we move the call to `inited` out of `calculate_workspace_edit` so we don't need to keep `ctx_` around?